### PR TITLE
Add aws e2e test

### PR DESCRIPTION
**Pull request by eversC**
this PR adds a `e2e_test_aws` job to CircleCI

the job itself builds the cloud-key-rotator binary, downloads some config from AWS, then runs the binary with the `rotate` cmd

I also de-duped the workflow section in the CircleCI config. Separate (but virtually identical) workflows had been defined with "ignore: master" and "only:master"

### CODE REVIEW

**Comment by chris-brindley** [comment]
Yay tests. Is there anything we can do to verify that the rotation has been successful within the test / CI pipeline? Just to pick up any bugs that cause the CKR to report that it is rotating but in fact doesn't actually do anything. Maybe another aws command after the rotation to confirm that the old credentials no longer work but the new ones do?

**Comment by eversC** [comment]
> Yay tests. Is there anything we can do to verify that the rotation has been successful within the test / CI pipeline? Just to pick up any bugs that cause the CKR to report that it is rotating but in fact doesn't actually do anything. Maybe another aws command after the rotation to confirm that the old credentials no longer work but the new ones do?

That's a great idea @chris-brindley, I'll take a look in my next 10%

File: .circleci/config.yml [hunk 1]
```diff
@@ -167,6 +167,27 @@ jobs:
              export GO111MODULE=on
              go mod download
              go test ./... -v
+
+  e2e_test_aws:
+    <<: *defaults
+
+    docker:
+      - image: eversc/aws-cli
```
Comment (enicholson125) [inline .circleci/config.yml:175]: It might be better to use the official `amazon/aws-cli` image, from a maintainability point of view?

**Review by enicholson125 — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (enicholson125) [reply .circleci/config.yml:175]: (Although then you'd need to install go...)

Comment (chris-brindley) [reply .circleci/config.yml:175]: Or you could do it the other way around and use an official go image and the install command from `circleci/aws-cli`?

Comment (chris-brindley) [reply .circleci/config.yml:175]: (Ah but then you might need to install python...) 😆

Comment (eversC) [reply .circleci/config.yml:175]: it was a real pain with either of those options

I think it'd be good to fork my `aws-cli` to the ovotech org, another 10% task I expect as there's a bit of setup involved

**Pull request merged**